Sources/protoc-gen-swift/ReservedWords.swift

```diff
@@ -47,47 +46,45 @@ private let reservedTypeNames: Set<String> = {
     return names
 }()
 
-func sanitizeMessageTypeName(_ s: String) -> String {
+private func sanitizeTypeName(_ s: String, disambiguator: String) -> String {
     if reservedTypeNames.contains(s) {
-        return s + "Message"
+        return s + disambiguator
     } else if isAllUnderscore(s) {
-        return s + "Message"
+        return s + disambiguator
+    } else if s.hasSuffix(disambiguator) {
+        let e = s.index(s.endIndex, offsetBy: -disambiguator.characters.count)
```

Comment:
maybe a comment on what this one is covering so when someone stumbles on it they don't have to figure it out?